src/contracts/nft/nft/src/lib.rs

```diff
@@ -127,37 +172,50 @@ impl Contract {
     #[payable]
     pub fn mint_tweet_request(
         &mut self,
-        tweet_id: String,
-        image_url: String,
-        notify: String,
+        input_data: String,
+        signature: Vec<u8>,
     ) -> (AccountId, u64, String) {
+        let input: InputMintRequest = serde_json::from_str(&input_data).unwrap();
+        let signature = Signature::from_slice(&signature.as_slice());
+        require!(signature.is_ok(), "Invalid signature");
+        let signature = signature.unwrap();
+        let verifying_key = VerifyingKey::from_sec1_bytes(self.agent_verification_key.as_slice())
+            .ok()
+            .expect("verifying_key must be correct");
+
         require!(
-            env::attached_deposit().eq(&MIN_DEPOSIT),
+            env::attached_deposit()
+                .ge(&(MIN_DEPOSIT + self.compute_cost(input.public_metric.clone()))),
```

Comment:
This is not calculated correctly.

The `compute_cost` evaluates points that must then multiply against some base rate per point.

Reply:
`&(MIN_DEPOSIT + self.compute_cost(input.public_metric.clone()))`
This is a sum between the points and the base costs,  assuming base cost is the `MIN_DEPOSIT`
That is incorrect.

It should be
`&(MIN_DEPOSIT * self.compute_cost(input.public_metric.clone()))`
Or
`&(BASE_COST_PER_POINT * self.compute_cost(input.public_metric.clone()))`